### PR TITLE
Ipld append car

### PR DESCRIPTION
**Pull request by turuslan**
- file merge sort to fit max_memory
- sparse index to fit max_memory
- truncate incomplete car
- rebuild outdated index

### CODE REVIEW

File: core/storage/car/cids_index/cids_index.cpp [hunk 1]
```diff
@@ -186,6 +412,24 @@ namespace fc::storage::cids_index {
...
+    if (auto key{asBlake(cid)}) {
+      if (writable) {
+        std::unique_lock lock{written_mutex};
+        Buffer item;
+        car::writeItem(item, cid, value);
+        Row row;
+        row.key = *key;
+        row.offset = car_offset;
+        row.max_size64 = maxSize64(item.size());
+        // TODO(turuslan): flush
```
Comment (Alexey-N-Chernyshov) [inline core/storage/car/cids_index/cids_index.cpp:424]: Why do not implement flush now?

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: core/storage/car/cids_index/cids_index.cpp [hunk 2]
```diff
@@ -11,82 +11,186 @@
...
+    auto read_error{ERROR_TEXT("merge: read error")};
+    auto write_error{ERROR_TEXT("merge: write error")};
+    std::greater<MergeRange> cmp;
+    for (auto it{ranges.begin()}; it != ranges.end();) {
+      auto &range{*it};
+      if (range.empty()) {
+        it = ranges.erase(it);
+      } else {
+        // estimated, 64kb
+        range.rows.resize(1638);
```
Comment (ortyomka) [inline core/storage/car/cids_index/cids_index.cpp:121]: It's not 64 kb

File: test/core/storage/car/cids_index_test.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      cids_path = car_path + ".cids";
+    }
+
+    auto load(bool writable) {
+      return loadOrCreateWithProgress(
+          car_path, writable, boost::none, nullptr, nullptr);
+    }
+  };
+
+  TEST_F(CidsIndexTest, Test) {
```
Comment (ortyomka) [inline test/core/storage/car/cids_index_test.cpp:30]: Please add some annotattion

**Review by ortyomka — APPROVED** [review]

File: core/storage/car/cids_index/cids_index.cpp (hunk 2 above)
Comment (turuslan) [reply core/storage/car/cids_index/cids_index.cpp:121]: `1638*sizeof(Row) = 1638*40 = 65520 = 64kb - 16b`

**Pull request merged**